rfc-108-including-gem-component-assets.md

```diff
@@ -0,0 +1,189 @@
+# Include specific component assets in applications
+
+## Problem and background
+
+The [current method for including the components](https://github.com/alphagov/govuk_publishing_components/blob/master/docs/install-and-use.md#4-include-the-assets) in a frontend application (such as `government-frontend`) is to include `_all_components.scss` and `_all_components_print.scss` in the Sass of that application. This has some benefits:
+
+- all components are automatically included in the application
+
+This also has some detriments:
+
+- all components are included regardless of whether they are used, so unused CSS/JS may be included
+- only publically facing components can be added to the gem in order to prevent the previous problem
```

Comment:
As discussed in the last `#govuk-frontenders` meeting we should avoid differentiating components based on where they are used. We can potentially rephrase that as: we are currently reserved in adding new components to the gem as we are concern about the performance of the public facing/frontend applications.